test/functional/test_framework/ripemd160.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2021 Pieter Wuille
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test-only pure Python RIPEMD160 implementation."""
```

Comment:
What makes this inherently test-only?

Reply:
It's not constant time.

Reply:
IMO should be mentioned in the comment.

Reply:
That's indeed a good warning to add in a followup.

Reply:
> It's not constant time.

Given that its only use in Bitcoin is `ripemd160(sha256(data))`, does this not make a timing attack effectively useless?

Reply:
@mmgen Yes, but it's open-source code, and open-source code gets copied and can end up being adopted for other purposes.